redis/tests/support/cluster.rs

```diff
@@ -118,77 +120,111 @@ impl RedisCluster {
                 mtls_enabled,
                 modules,
                 |cmd| {
-                    let tempdir = tempfile::Builder::new()
-                        .prefix("redis")
-                        .tempdir()
-                        .expect("failed to create tempdir");
-                    let acl_path = tempdir.path().join("users.acl");
-                    let acl_content = format!(
-                        "user {} on allcommands allkeys >{}",
-                        Self::username(),
-                        Self::password()
-                    );
-                    std::fs::write(&acl_path, acl_content).expect("failed to write acl file");
-                    cmd.arg("--cluster-enabled")
-                        .arg("yes")
-                        .arg("--cluster-config-file")
-                        .arg(&tempdir.path().join("nodes.conf"))
-                        .arg("--cluster-node-timeout")
-                        .arg("5000")
-                        .arg("--appendonly")
-                        .arg("yes")
-                        .arg("--aclfile")
-                        .arg(&acl_path);
-                    if is_tls {
-                        cmd.arg("--tls-cluster").arg("yes");
-                        if replicas > 0 {
-                            cmd.arg("--tls-replication").arg("yes");
+                    let mut cur_attempts = 0;
+                    loop {
+                        let tempdir = tempfile::Builder::new()
+                            .prefix("redis")
+                            .tempdir()
+                            .expect("failed to create tempdir");
+                        let acl_path = tempdir.path().join("users.acl");
+                        let acl_content = format!(
+                            "user {} on allcommands allkeys >{}",
+                            Self::username(),
+                            Self::password()
+                        );
+                        std::fs::write(&acl_path, acl_content).expect("failed to write acl file");
+                        cmd.arg("--cluster-enabled")
```

Comment:
you add args to `cmd` in every loop, which sounds incorrect. shouldn't the loop only encompass `cmd.spawn`? or maybe create a base `cmd` before the loop, and in the loop only create the tempdir and add it to a clone of `cmd`?